text/3263-precise-pre-release-deps.md

```diff
@@ -0,0 +1,141 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The following assumes:
+
+ - A release is a version without a "pre-release" tag (like `1.0.0`)
+ - A pre-release is a version with a "pre-release" tag (like `1.0.0-alpha`)
+
+By default Cargo use the more compatible version for release of a crate but will not for pre-release.
+
+```toml
+[dependencies.foo]
+version = "1.0.0-alpha.0"
+```
+
+will now be considered by Cargo as:
+
+```toml
+[dependencies.foo]
+version = "=1.0.0-alpha.0"
+```
+
+while release version like:
+
+```toml
+[dependencies.foo]
+version = "1.0.0"
+```
+
+are considered by Cargo as:
+
+```toml
+[dependencies.foo]
+version = "^1.0.0"
+```
+
+The latter doesn't change.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Do it transparently for the user.
+
+Cargo will need to differentiate how they resolve a pre-release and a release version.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+ * This could break some build. A code that was using a most recent pre-release version could not compile anymore.
+ * It's make Cargo have a different behavior for version field between a pre-release and a release. Making both maintenance of Cargo harder but also could confuse user of Rust.
+ * It's a change that affect the whole Rust ecosystem.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+In practice, do it transparently should not break any heathy workflow, user that put a pre-release version clearly want to use this version and if user have been using a most recent pre-release version not on purpose it's very unlikely their build would break (build would have break when new pre release was out). We could check the impact of implementing this transparently with a [crater](https://github.com/rust-lang/crater) run. If we see that there are unacceptable break, This RFC will then say we will use an opt-in option as follow:
+
+  1. Use the `resolver` value version, bump it to `3`:
+   
+     This may be overkill but it's a naturel way to make change in the way semver rule are treated by Cargo.
+     
+  2. Use a separate value `pre-release-updates = "sticky" # or "default"`:
+
+     This is less overkill and can be remove at the next Edition of Rust.
+
+Instead of changing resolver cargo behavior, we could decide that there is no compatible version for pre-release as explaining pre-release having compatible version don't make a lot of sense. So `1.0.0-alpha.0` would never match any other requirement that exact same version. That mean that `^1.0.0-alpha.0` could only match `1.0.0-alpha.0` version. This have the major benefit to not introduce inconsistency with pre-release and release in Cargo resolve. This could also be adopted transparently or using opt-in solution.
```

Comment:
This might be a typo?

```suggestion
Instead of changing resolver cargo behavior, we could decide that there is no compatible version for pre-release as explaining pre-release having compatible version don't make a lot of sense. So `1.0.0-alpha.0` would never match any other requirement than that exact same version. That mean that `^1.0.0-alpha.0` could only match `1.0.0-alpha.0` version. This have the major benefit to not introduce inconsistency with pre-release and release in Cargo resolve. This could also be adopted transparently or using opt-in solution.
```

Reply:
Sorry I have dyslexia and often do such mistake feel free to fix anything you find, thus, I expect I did a lot of small mistake so maybe this could be on the form of a big commit.